PolyEngine/Engine/Src/Imgui/ImguiSystem.cpp

```diff
@@ -48,118 +121,63 @@ void ImguiSystem::ImguiUpdatePhase(Scene* scene)
-// 	SDL_CaptureMouse(any_mouse_button_down ? SDL_TRUE : SDL_FALSE);
-// #else
-	// if (SDL_GetWindowFlags(g_Window) & SDL_WINDOW_INPUT_FOCUS)
+	// Set OS mouse position if requested (rarely used, only when ImGuiConfigFlags_NavEnableSetMousePos is enabled by user)
+	if (io.WantSetMousePos)
+		gEngine->GetOutputQueue().PushBack({eOutputEventType::MOUSEPOS, Vector2i((int)io.MousePos.x, (int)io.MousePos.y)});
+	else
+		io.MousePos = ImVec2(-FLT_MAX, -FLT_MAX);
+
 	Vector2i mousePos = inputCmp->GetMousePos();
 	io.MousePos = ImVec2((float)mousePos.X, (float)mousePos.Y);
-// #endif
-// }
-
-
-	if (io.ConfigFlags & ImGuiConfigFlags_NoMouseCursorChange)
-		return;
-
-	// ImGuiMouseCursor imgui_cursor = ImGui::GetMouseCursor();
-	// if (io.MouseDrawCursor || imgui_cursor == ImGuiMouseCursor_None)
-	// {
-	// 	Hide OS mouse cursor if imgui is drawing it or if it wants no cursor
-	// 	SDL_ShowCursor(SDL_FALSE);
-	// 	inputCmp->SetCursorHidden(); // TODO: input cursor visibility handling
-	// }
-	// else
-	// {
-	// 	Show OS mouse cursor
-	// 	SDL_SetCursor(g_MouseCursors[imgui_cursor] ? g_MouseCursors[imgui_cursor] : g_MouseCursors[ImGuiMouseCursor_Arrow]);
-	// 	SDL_ShowCursor(SDL_TRUE);
-	// }
 
+	if (!(io.ConfigFlags & ImGuiConfigFlags_NoMouseCursorChange))
+	{
+		ImGuiMouseCursor imgui_cursor = ImGui::GetMouseCursor();
+		if (io.MouseDrawCursor || imgui_cursor == ImGuiMouseCursor_None)
+		{
+			// Hide OS mouse cursor if imgui is drawing it or if it wants no cursor
+			gEngine->GetOutputQueue().PushBack({ eOutputEventType::CURSORHIDE });
+		}
+		else
+		{
+			// Show OS mouse cursor
+			gEngine->GetOutputQueue().PushBack({ eOutputEventType::CURSORSET, GetCursorType(imgui_cursor) });
+			gEngine->GetOutputQueue().PushBack({ eOutputEventType::CURSORSHOW });
+		}
+	}
 
 	IM_ASSERT(io.Fonts->IsBuilt());     // Font atlas needs to be built, call renderer _NewFrame() function e.g. ImGui_ImplOpenGL3_NewFrame()
-										// Setup display size (every frame to accommodate for window resizing)
-	int w, h;
-	int display_w, display_h;
-	// SDL_GetWindowSize(window, &w, &h);
+	
+	// Setup display size (every frame to accommodate for window resizing)
 	ScreenSize screenSize = gEngine->GetRenderingDevice()->GetScreenSize();
-	display_w = w = screenSize.Width;
-	display_h = h = screenSize.Height;
-
-	// SDL_GL_GetDrawableSize(window, &display_w, &display_h);
-	io.DisplaySize = ImVec2((float)w, (float)h);
-	io.DisplayFramebufferScale = ImVec2(
-		w > 0 ? ((float)display_w / w) : 0,
-		h > 0 ? ((float)display_h / h) : 0
-	);
-
-	// Setup time step (we don't use SDL_GetTicks() because it is using millisecond resolution)
-	// static Uint64 frequency = SDL_GetPerformanceFrequency();
-	// Uint64 current_time = SDL_GetPerformanceCounter();
-	// io.DeltaTime = g_Time > 0 ? (float)((double)(current_time - g_Time) / frequency) : (float)(1.0f / 60.0f);
+	io.DisplaySize = ImVec2((float)screenSize.Width, (float)screenSize.Height);
+	io.DisplayFramebufferScale = ImVec2(1.0f, 1.0f);
```

Comment:
Maybe we can create a constant in some kind of unified place, for RenderView parameters? We can set that to one, but in the future, viewport scale would be nice. I guess this is important for future rendering refactor.

Reply:
RenderViewSettings accessed through CameraComponent would be useful both in UI/Editor and Rendering.

Would like to start with reported test case when scaling is critically needed ex. dpi scaling on some devices. I would be easier to test how scaling is behaving against concrete test scenario.

Nice thing: Spotted that Imgui remembers panel positions in window between .exe runs. If panel is left in fullscreen otside of window area it's position is clipped to window bounds.